objcgen/objcgenerator-processor.cs

```diff
@@ -253,77 +253,83 @@ public override void Process (IEnumerable<Assembly> input)
 				if (!AddIfUnique (pa))
 					throw ErrorHelper.CreateError (12, $"The assembly name `{pa.Name}` is not unique");
 
-				foreach (var t in GetTypes (a)) {
-					if (t.IsEnum) {
-						enums.Add (new ProcessedType (t));
-						continue;
-					}
+				foreach (var t in GetTypes (a))
+					Process (t);
+			}
+			types = types.OrderBy ((arg) => arg.Type.FullName).OrderBy ((arg) => types.Contains (arg.Type.BaseType)).ToList ();
 
-					if (t.IsInterface) {
-						protocols.Add (new ProcessedType (t));
-					} else {
-						types.Add (new ProcessedType (t));
-					}
+			ErrorHelper.Show (delayed);
+		}
 
-					extension_type = t.HasCustomAttribute ("System.Runtime.CompilerServices", "ExtensionAttribute");
-
-					implement_system_icomparable = t.Implements ("System", "IComparable");
-					implement_system_icomparable_t = t.Implements("System", "IComparable`1");
-
-					var constructors = GetConstructors (t).OrderBy ((arg) => arg.ParameterCount).ToList ();
-					var processedConstructors = PostProcessConstructors (constructors).ToList ();
-					if (processedConstructors.Count > 0)
-						ctors.Add (t, processedConstructors);
-
-					var typeEquals = equals.Where (x => x.Key == t).Select (x => x.Value);
-					var meths = GetMethods (t).OrderBy ((arg) => arg.Name).ToList ();
-					var processedMethods = PostProcessMethods (meths, typeEquals).ToList ();
-					if (processedMethods.Count > 0)
-						methods.Add (t, processedMethods);
-
-					var props = new List<PropertyInfo> ();
-					var subscriptProps = new List<PropertyInfo> ();
-					foreach (var pi in GetProperties (t)) {
-						var getter = pi.GetGetMethod ();
-						var setter = pi.GetSetMethod ();
-						// setter only property are valid in .NET and we need to generate a method in ObjC (there's no writeonly properties)
-						if (getter == null)
-							continue;
-
-						// indexers are implemented as methods and object subscripting
-						if ((getter.ParameterCount > 0) || ((setter != null) && setter.ParameterCount > 1)) {
-							subscriptProps.Add (pi);
-							continue;
-						}
+		public void Process (Type t, Type outerType = null)
```

Comment:
Knew there had to be a way to get this from the Type, but missed DeclaringType in there...updating.